transport/peer/peerlist/single_test.go

```diff
@@ -0,0 +1,211 @@
+package peerlist
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"go.uber.org/yarpc/transport"
+	"go.uber.org/yarpc/transport/internal/errors"
+	"go.uber.org/yarpc/transport/transporttest"
+
+	"github.com/crossdock/crossdock-go/assert"
+	"github.com/golang/mock/gomock"
+)
+
+func TestSingle(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	type expectedChooseResult struct {
+		peer transport.Peer
+		err  error
+	}
+
+	type testStruct struct {
+		msg                   string
+		pid                   transport.PeerIdentifier
+		agent                 *transporttest.MockPeerAgent
+		appliedFunc           func(*single) error
+		expectedPeerID        transport.PeerIdentifier
+		expectedPeer          transport.Peer
+		expectedAgent         transport.PeerAgent
+		expectedStarted       bool
+		expectedErr           error
+		expectedChooseResults []expectedChooseResult
+	}
+	tests := []testStruct{
+		func() (s testStruct) {
+			s.msg = "setup"
+			s.pid = transporttest.NewMockPeerIdentifier(mockCtrl)
+			s.agent = transporttest.NewMockPeerAgent(mockCtrl)
+
+			s.appliedFunc = func(pl *single) error {
+				return nil
+			}
+
+			s.expectedPeerID = s.pid
+			s.expectedAgent = s.agent
+			s.expectedStarted = false
+			return
```

Comment:
Instead of combining all these tests, it's okay (preferable even) to split
tests for different behaviors out into different test functions. Table tests
are for tests with enough similar structure that writing those into separate
functions would be redundant. As it stands right now, it's very difficult to
figure out what each of these tests is actually testing.